staff/views/applications.py

```diff
@@ -31,10 +32,15 @@ def post(self, request, *args, **kwargs):
         wizard = Wizard(application, form_specs)
         pages = list(wizard.get_pages())
 
+        review_time = timezone.now()
+
         for page in pages:
             form = page.get_bound_approval_form(request.POST)
-            if form.instance.pk:
+            if form.instance.pk and form.has_changed():
                 form.save()
+                page.instance.last_reviewed_at = review_time
+                page.instance.reviewed_by = request.user
+                page.instance.save()
```

Comment:
I'd prefer it if we had `form.instance` here instead of `page.instance` -- although they're the same object, that isn't obvious from reading this code.